src/transformers/models/mamba2/modeling_mamba2.py

```diff
@@ -1893,3 +1901,163 @@ def forward(
             hidden_states=outputs.hidden_states,
             attentions=outputs.attentions,
         )
+
+
+class Mamba2ClassificationHead(nn.Module):
+    """Head for sentence-level classification tasks."""
+
+    def __init__(self, config):
+        """Initialize the head."""
+        super().__init__()
+        self.activation = ACT2FN[config.hidden_act]
+        self.dense = nn.Linear(config.hidden_size, config.hidden_size)
+        self.dropout = nn.Dropout(config.classifier_dropout)
+        self.out_proj = nn.Linear(config.hidden_size, config.num_labels)
+
+        self.config = config
+
+    def forward(self, inputs, **kwargs):
+        """Forward pass."""
+        # Pooling is done by the forward pass in `Mamba2ForSequenceClassification`
+        hidden_states = self.dropout(inputs)
+        hidden_states = self.dense(hidden_states)
+        hidden_states = self.activation(hidden_states)
+        hidden_states = self.dropout(hidden_states)
+
+        return self.out_proj(hidden_states)
```

Comment:
Can be copied from Bart at least, no? Would need slight adjustments in the model below but nothing major (e.g. init should also be copiable from Bart, forward would only need a few renames).